v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -379,7 +411,7 @@ async function parseNamedParameters(
     ) {
```

Comment:
same here, you don't need the `usedCliArguments[i + 1] !== undefined` check

Reply:
N/A, see https://github.com/NomicFoundation/hardhat/pull/5334#discussion_r1628104331